xpath/src/main/scala/advxml/xpath/error/Error.scala

```diff
@@ -0,0 +1,11 @@
+package advxml.xpath.error
+
+import cats.parse.Parser
+import eu.cdevreeze.xpathparser.ast.XPathElem
+
+sealed trait Error
+final case class ParsingError(err: Parser.Error) extends Error {
+  override def toString: String =
+    s"Parsing failed at ${err.failedAtOffset}: ${err.expected.toList.mkString}"
+}
+final case class NotSupportedConstruction(feature: XPathElem) extends Error
```

Comment:
I would put the coproduct instances into the companion object.
I wonder if `Error` is fine or we should name this `XPathError`, what do you think ? 

```suggestion
sealed trait XPathError
object XPathError {
  final case class ParsingError(err: Parser.Error) extends XPathError {
    override def toString: String =
      s"Parsing failed at ${err.failedAtOffset}: ${err.expected.toList.mkString}"
  }
  final case class ConstructionError(message: String) extends XPathError {
    override def toString: String = message
  }
}
```